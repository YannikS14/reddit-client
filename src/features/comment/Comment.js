import moment from 'moment';
import React from 'react';

export default function Comment({ comment }) {
  const commentDate = moment(comment.data.created * 1000).fromNow();

  return (
    <div
      id="comment"
      className="mb-2 px-8 py-3 bg-gray-100 dark:bg-gray-600 rounded-md hover:shadow"
    >
      <div className="flex justify-between pb-2">
        <div className="font-semibold text-sm">
          {comment.data.author}
        </div>
        <div className="text-xs">{commentDate}</div>
      </div>
      <p>{comment.data.body}</p>
    </div>
  );
}
