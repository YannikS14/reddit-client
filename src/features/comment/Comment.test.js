import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Comment from './Comment';

describe('Comment component', () => {
  const testComment = {
    data: {
      created: 1626123217,
      author: 'Testauthor',
      body: 'This is a test comment body.',
    },
  };

  it('renders the creation date of the comment', () => {
    const wrapper = shallow(<Comment comment={testComment} />);
    const text = wrapper.find('div.text-xs').text();
    expect(text).toEqual('4 months ago');
  });

  it('renders the author of the comment', () => {
    const wrapper = shallow(<Comment comment={testComment} />);
    const text = wrapper.find('div.font-semibold.text-sm').text();
    expect(text).toEqual(testComment.data.author);
  });

  it('renders the body of the comment', () => {
    const wrapper = shallow(<Comment comment={testComment} />);
    const text = wrapper.find('p').text();
    expect(text).toEqual(testComment.data.body);
  });

  it('matches the snapshot', () => {
    const tree = renderer
      .create(<Comment comment={testComment} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
