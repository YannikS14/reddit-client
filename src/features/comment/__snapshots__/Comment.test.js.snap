// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment component matches the snapshot 1`] = `
<div
  className="mb-2 px-8 py-3 bg-gray-100 dark:bg-gray-600 rounded-md hover:shadow"
  id="comment"
>
  <div
    className="flex justify-between pb-2"
  >
    <div
      className="font-semibold text-sm"
    >
      Testauthor
    </div>
    <div
      className="text-xs"
    >
      4 months ago
    </div>
  </div>
  <p>
    This is a test comment body.
  </p>
</div>
`;
